presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -606,15 +612,55 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
 
         // TODO Adjust statistics to take into account required subfields
 
-        Map<String, ColumnHandle> columns = columnHandles.stream()
+        if (!tableLayoutHandle.isPresent() || !((HiveTableLayoutHandle) tableLayoutHandle.get()).isPushdownFilterEnabled()) {
```

Comment:
@mbasmanova  has an answer to this

Reply:
@rschlussel Rebecca, this check is not using session property, but rather a property in the layout handle itself. That property indicates that the filter was indeed pushed down. If the filter wasn't pushed down, the layout still have a piece of it (used for stripe pruning based on file stats). In that case it would be incorrect to prune stats using that filter, because they'll be pruned again by the FilterNode.